sw/device/lib/base/status.h

```diff
@@ -56,6 +56,53 @@ typedef struct status {
   int32_t value;
 } status_t;
 
+/**
+ * Record the creation of a status_t in a special section of the executable.
+ * This can be used by tools to more easily decode status_t values.
+ */
+#define RECORD_STATUS_CREATE(code, mod_id, file) \
+  _RECORD_STATUS_CREATE(code, mod_id, file)
+
+/* Multiple instances of this structure will go to the .ot.status_create_record
+ * section and will be read by tools. Therefore it is important that the layout
+ * of this structure remains stable. */
+typedef struct ot_status_create_record {
+  // Set to OT_SCR_UNKNOWN_MOD_ID if not fixed at compile time
+  uint16_t module_id;
+  char filename[126];
+} __attribute__((packed)) ot_status_create_record_t;
```

Comment:
I want it to be packed because it is then exported to rust with bindgen. The only goal of this data structure is to store data in an ELF file using a well-defined format, it does not serve any other purpose and therefore should not have gaps.

Reply:
C struct layouts are already well-defined. If all fields are naturally aligned, then there will be no gaps. The assertion serves as the purpose to ensure that (if the total size is equal to the sum of size of fields, then there couldn't be any gaps).

The fact that it's exposed via bindgen is the reason that I'm against using `packed`, because in general packed structs are a nightmare in bindgen and Rust and it's best to avoid.

Reply:
I am happy to remove the `packed` if you feel so strongly about it. As I said, this is a file storage structure, the fact that I define in C is just convenience. Using static assert and C aligment rules defeats the entire purpose of having a compiler feature to achieve what I want and express in the most natural way.

Reply:
If you are using C ABI as interop tool, then the most portable way is to avoid `packed` (so it translates to Rust `#[repr(C)]`) and bitfields. (Actually, it's best to avoid having any attributes at all)